ICSharpCode.Decompiler/CSharp/CallBuilder.cs

```diff
@@ -1232,7 +1232,17 @@ TranslatedExpression HandleDelegateConstruction(CallInstruction inst)
 				default:
 					throw new ArgumentException($"Unknown instruction type: {func.OpCode}");
 			}
-			var invokeMethod = inst.Method.DeclaringType.GetDelegateInvokeMethod();
+			return HandleDelegateConstruction(inst.Method.DeclaringType, method, expectedTargetDetails, thisArg, inst);
+		}
+
+		internal TranslatedExpression Build(LdVirtDelegate inst)
+		{
+			return HandleDelegateConstruction(inst.Type, inst.Method, new ExpectedTargetDetails { CallOpCode = OpCode.CallVirt }, inst.Argument, inst);
+		}
+
+		TranslatedExpression HandleDelegateConstruction(IType delegateType, IMethod method, ExpectedTargetDetails expectedTargetDetails, ILInstruction thisArg, ILInstruction inst)
+		{
+			var invokeMethod = delegateType.GetDelegateInvokeMethod();
```

Comment:
If the invoke method is necessary, shouldn't we store it with the `LdVirtDelegate` instruction?
For `UnknownType`, it's not possible to retrieve the Invoke `FakeMethod` from the type.

Reply:
Ah but we need to go from ctor to invoke, so we have no chance to discover the signature of an UnknownType delegate.